packages/components/src/components/CodeEditor.tsx

```diff
@@ -18,59 +49,162 @@ interface CodeEditorProps {
-      height={String(height) + "px"}
-      minLines={oneLine ? lineCount : undefined}
-      maxLines={oneLine ? lineCount : undefined}
-      showGutter={showGutter}
-      highlightActiveLine={false}
-      showPrintMargin={false}
-      onChange={onChange}
-      name={id}
-      editorProps={{
-        $blockScrolling: true,
-      }}
-      setOptions={{}}
-      commands={[
-        {
-          name: "submit",
-          bindKey: { mac: "Cmd-Enter", win: "Ctrl-Enter" },
-          exec: () => onSubmitRef.current?.(),
-        },
-      ]}
-      markers={errorLocations?.map((location) => ({
-        startRow: location.start.line - 1,
-        startCol: location.start.column - 1,
-        endRow: location.end.line - 1,
-        endCol: location.end.column - 1,
-        className: "ace-error-marker",
-        type: "text",
-      }))}
-    />
+  const editor = useRef<HTMLDivElement>(null);
+  const editorView = useRef<EditorView | null>(null);
+  const languageSupport = useMemo(squiggle, []);
+  const state = useMemo(
+    () =>
+      EditorState.create({
+        doc: value,
+        extensions: [
+          highlightSpecialChars(),
+          history(),
+          drawSelection(),
+          dropCursor(),
+          EditorState.allowMultipleSelections.of(true),
+          indentOnInput(),
+          syntaxHighlighting(lightThemeHighlightingStyle, { fallback: true }),
+          bracketMatching(),
+          closeBrackets(),
+          autocompletion(),
+          // rectangularSelection(),
+          // crosshairCursor(),
+          highlightSelectionMatches({
+            wholeWords: true,
+            highlightWordAroundCursor: false, // Weird on numbers
+          }),
+          keymap.of([
+            ...closeBracketsKeymap,
+            ...defaultKeymap,
+            ...searchKeymap,
+            ...historyKeymap,
+            ...foldKeymap,
+            ...completionKeymap,
+            ...lintKeymap,
+            indentWithTab,
+          ]),
+          compGutter.of([]),
+          compUpdateListener.of([]),
+          compSubmitListener.of([]),
+          compTheme.of([]),
+          languageSupport,
+        ],
+      }),
+    []
   );
+
+  useEffect(() => {
+    if (editor.current != null && state != null) {
+      const view = new EditorView({ state, parent: editor.current });
+      editorView.current = view;
+
+      return () => {
+        view.destroy();
+      };
+    }
+  }, [state]);
+
+  useEffect(() => {
+    editorView.current?.dispatch({
+      effects: compGutter.reconfigure(
+        showGutter
+          ? [
+              lineNumbers(),
+              highlightActiveLine(),
+              highlightActiveLineGutter(),
+              foldGutter(),
+            ]
+          : []
+      ),
+    });
+  }, [showGutter]);
+
+  useEffect(() => {
+    editorView.current?.dispatch({
+      effects: compUpdateListener.reconfigure(
+        EditorView.updateListener.of((update) => {
+          // console.log(
```

Comment:
Should you remove this commented code?

Reply:
It's something that really helps debugging the parser — so I would like to leave a documented comment there~